### DISCUSSION TITLE
Tokenizing payment is not throwing error but is also not updating the cart with billing info

### DISCUSSION THREAD

**Discussion by Buupu** [swellstores/community#70] [Help] [answered]
I've followed the Stripe integration outlined [here ](https://developers.swell.is/frontend-api/payments) in the Swell docs but when the following code is executed
```
await swell.payment.tokenize({
    card: {
      onError: (err) => {
        // inform the customer there was an error
      },
      onSuccess: () => {
        //finally submit the form
        form.submit();
      }
    }
    // ideal: { onError: (err) => {}, ...}
    // klarna: { onError: (err) => {}, ...}
    // bancontact: { onError: (err) => {}, ...}
 });
```
  
Neither the onError nor the onSuccess handlers are triggered and the billing info is not applied to the basket. Having looked through the source code I think the issue might lie [here](https://github.com/swellstores/swell-js/blob/fef31353fabce391cca56c31bb2a57b8ebf19560/src/payment.js#L539).
  
Swell is expecting the intent status to be `requires_confirmation` but the status is being returned as `requires_capture`. Therefore, the if statement is never entered meaning no billing info to the basket, and no error is thrown.

I'd really appreciate any help with this!

**Comment by logeshswell** [comment]
It seems like this issue is interconnected to this [discussion](https://github.com/swellstores/community/discussions/66).
~So, I am marking it as duplicate label.~

**Reply by Buupu** [reply]
I have participated in the discussion youre referencing. They're separate issues.

**Reply by Buupu** [reply]
One issue relates to the use of camelCase/snake case and the other relates to swell js not handling the Stripe intent status correctly.

**Reply by logeshswell** [reply]
@Buupu Thanks for the heads up!

**Comment by logeshswell** [comment, marked as answer]
This issue is resolved as part of recent swell-js release [v3.18.3](https://github.com/swellstores/swell-js/releases/tag/v3.18.3)
